File: EuclidsAlgorithm/go/euclid.go
/*
	Package euclid allows for finding the greatest common denominator
	between two numbers.
*/
package euclid

// Function euclid finds the greates common denominator between two unsigned
// integers m and n. If either value is 0 the result will be 0.
func euclid(m,n uint) (uint) {
	var r uint
	if(m==0||n==0) {return 0}
	for {
		r = m%n
		if(r==0) {break}
		m = n
		n = r
	}
	return n
}

File: EuclidsAlgorithm/javascript/Euclid.js
/**
 * Euclid's algorithm can be used to find the greatest commmon denominator
 * between to input values. This function will return 0 if either input is
 * 0 or -1 if either input is negative.
 */
function Euclid(input1,input2) {
	if(input1==0||input2==0){return 0;}
	if(input1<0||input2<0){return -1;}
	while(true) {
		remainder=input1%input2;
		if(remainder==0) {return input2;}
		input1 = input2;
		input2 = remainder;
	}
}
